### PR TITLE
CS audit changes

### PR DESCRIPTION
**Pull request by talbaneth**

### CODE REVIEW

File: src/DssProxyActionsCharter.sol [hunk 1]
```diff
@@ -372,12 +377,11 @@ contract DssProxyActionsCharter is Common {
         bytes32 ilk,
         address daiJoin
     ) external {
-        address vat = CharterLike(charter).vat();
         address urp = CharterLike(charter).getOrCreateProxy(address(this));
         (, uint256 art) = VatLike(vat).urns(ilk, urp);
 
         // Joins DAI amount into the vat
-        daiJoin_join(daiJoin, _getWipeAllWad(vat, urp, ilk));
+        daiJoin_join(daiJoin, _getWipeAllWad(urp, ilk));
```
Comment (kmbarry1) [inline src/DssProxyActionsCharter.sol:384]: Do we maybe want to optimize for not doing a join if `_getWipeAllWad` returns zero? I guess it is an amortization question that does not have an obvious answer, so fine either way.

**Review by kmbarry1 — APPROVED** [review]
LGTM

Don't see anything that requires a fix, left one comment to consider.

**Comment by gbalabasquer** [comment]
I think we should also make the `charter` immutable as well. Maybe some other contract as well, but `charter` would make sense as does with the `vat`.

File: src/DssProxyActionsCharter.sol (hunk 1 above)
Comment (talbaneth) [reply src/DssProxyActionsCharter.sol:384]: I think the case where it returns zero is by far less likely, so I will not do this atm.

**Comment by talbaneth** [comment]
> I think we should also make the `charter` immutable as well. Maybe some other contract as well, but `charter` would make sense as does with the `vat`.

done

**Review by gbalabasquer — APPROVED** [review]
LGTM

**Review by kmbarry1 — APPROVED** [review]

**Pull request merged**